docs/data-sources/modelarts_datasets.md

```diff
@@ -0,0 +1,110 @@
+---
+subcategory: "Beta"
+---
+
+# huaweicloud_modelarts_datasets
+
+Use this data source to get a list of ModelArts datasets.
+
+## Example Usage
+
+```hcl
+data "huaweicloud_modelarts_datasets" "test" {
+  name = "your_dataset_name"
+  type = 1
+}
+```
+
+## Argument Reference
+
+The arguments of this data source act as filters for querying the available datasets in the current region.
+ All datasets that meet the filter criteria will be exported as attributes.
+
+* `region` - (Optional, String) Specifies the region in which to obtain datasets. If omitted, the provider-level region
+ will be used.
+
+* `name` - (Optional, String) Specifies the name of datasets.
+
+* `type` - (Optional, String) Specifies the type of datasets. The options are:
+  + **0**: Image classification, supported formats: `.jpg`, `.png`, `.jpeg`, `.bmp`.
+  + **1**: Object detection, supported formats: `.jpg`, `.png`, `.jpeg`, `.bmp`.
+  + **3**: Image segmentation, supported formats: `.jpg`, `.png`, `.jpeg`, `.bmp`.
+  + **100**: Text classification, supported formats: `.txt`, `.csv`.
+  + **200**: Sound classification, Supported formats: `.wav`.
+  + **400**: Table type, supported formats: Carbon type.
+  + **600**: Video, supported formats: `.mp4`
+  + **900**: Free format.
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - Indicates a data source ID.
+
+* `datasets` - Indicates a list of all datasets found. Structure is documented below.
+
+The `datasets` block contains:
+
+* `id` - The ID of the datasets.
```

Comment:
The `id` here should represent the ID of a single object and should be singular.